### DISCUSSION TITLE
Exploring the Feasibility of Simultaneously Viewing Multiple Screens in a Proctoring System

### DISCUSSION THREAD

**Discussion by gsabarinath02** [Jaideep25/Videolify#6] [General]
Hello,

I am currently in the process of developing a proctoring system, and I am interested in the possibility of displaying multiple persons' screens simultaneously. Specifically, I would like to be able to view the screens of several individuals simultaneously, similar to the functionality offered by Discord, such that upon entering a room, I am able to view all streams simultaneously on my system.

I am wondering if it is technically feasible to achieve this and if there are any applications available on GitHub that could serve as a reference for the implementation of such functionality. Any insights or recommendations would be greatly appreciated.

Thank you in advance.

**Comment by Jaideep25** [comment]
Hey @gsabarinath02 ,

My app couldn't be of any use in creating a proctored system. But, I do know some repositories,

1. By @miroslavpejic85 https://github.com/miroslavpejic85/mirotalksfu -> You can add a functionality to prompt to share their screens at the start of the meeting and add a text which shows "You stopped screen sharing", It can handle up to roughly 250-300 users (Capacity not tested personally) This would help for the video conferencing...

2. Check this out too - https://github.com/vardanagarwal/Proctoring-AI. This is for your eye tracking, gesture movement, etc...

3. I found this exciting project, https://github.com/tusharnankani/Aankh. Here you'd need to add the video conferencing part...


A suggestion ~ It would be better to use an AI over humans for proctored tests, if AI finds anything suspicious it would play the video on the admin's screen for monitoring purposes, instead of a human sitting in front of the system. 
Secondly, it'd be better if you go for face tracking (it would help if the user is using an extra gadget) and add a function to check if the user closes the proctored window or not, instead of staring at their screens.

I wish you good luck, on your journey in this project!
Regards ;)

**Reply by gsabarinath02** [reply]
Thank you so much for sharing those repositories with me. I will definitely check them out and see if they can be useful in our proctoring system.

Regarding your suggestion about using AI over humans for proctored tests, that's a great idea. We will definitely explore the possibility of integrating AI in our system. However, as you mentioned, for some exams, candidates need to switch between tabs for using software like Autocad, excel, etc. Our goal is not to restrict them from switching the screen, but to ensure that they are not cheating during the exam.

To address this, we are planning to implement a system where the invigilator can see all candidate screens at the same time. Do you have any suggestions on how we can achieve this? We want to ensure that the system is effective in detecting cheating, but also user-friendly for both the invigilator and the candidate.

Thank you again for your help and support.

Best regards.
